shim/linux/src/entry.c

```diff
@@ -66,6 +86,28 @@ dispatch_system_kvm_check_extension(void)
 static long
 dispatch_system_kvm_create_vm(void)
 {
+    char vmname[22];
+    struct shim_vm_t *vm =
+        (struct shim_vm_t *)vmalloc(sizeof(struct shim_vm_t));
+    if (NULL == vm) {
+        goto vm_free;
```

Comment:
before each goto, you should have a debug statement telling us what the error was. You can see how this is done in the shim_init.c file.